services/app-web/src/components/FilterAccordion/FilterSelect/FilterSelect.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useRef, useEffect } from 'react'
+import styles from '../../Select/Select.module.scss'
+import Select, {
+    AriaOnFocus,
+    Props,
+    SelectInstance,
+    GroupBase,
+    components,
+    MenuListProps,
+} from 'react-select'
+import { usePrevious } from '../../../hooks'
+
+export type FilterSelectPropType = {
+    name: string
+    initialValues?: string[]
+    filterOptions: FilterOptionType[]
+    label?: string
+    toggleClearFilter?: boolean
+}
+
+export type FilterOptionType = {
+    readonly label: string
+    readonly value: string
+}
+
+export const FilterSelect = ({
+    name,
+    initialValues,
+    filterOptions,
+    label,
+    toggleClearFilter,
+    ...selectProps
+}: FilterSelectPropType & Props<FilterOptionType, true>) => {
+    const prevToggleClearFilter = usePrevious(toggleClearFilter)
+    const selectInputRef =
+        useRef<
+            SelectInstance<FilterOptionType, true, GroupBase<FilterOptionType>>
+        >(null)
+
+    const onFocus: AriaOnFocus<FilterOptionType> = ({
+        focused,
+        isDisabled,
+    }): string => {
+        return `You are currently focused on option ${focused.label}${
+            isDisabled ? ', disabled' : ''
+        }`
+    }
+
+    //This component just wraps the menu list inside a div, so we can set a data-testid and run jest tests.
+    const MenuList = (props: MenuListProps<FilterOptionType>) => {
```

Comment:
nitpick: This could be declared outside the component to avoid unnecessary re-rendering. I only know this cuz we ran into some issues with cypress that were caused by sub components declared inside a component being re-rendered unnecessarily so now I make an effort to declare outside.